schema-registry-mock/src/test/java/com/bakdata/schemaregistrymock/ProtobufRegistryMockTest.java

```diff
@@ -0,0 +1,174 @@
+package com.bakdata.schemaregistrymock;
+
+import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchemaProvider;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ProtobufRegistryMockTest {
+    private final SchemaRegistryMock schemaRegistry = new SchemaRegistryMock(List.of(new ProtobufSchemaProvider()));
+    private final ParsedSchema schema = createSchema();
+
+    public ProtobufRegistryMockTest() throws IOException {
```

Comment:
what is this constructor for?

Reply:
`createSchema` throws an IOException and is called during object creation (line 24), therefore this was needed.

I moved the field initialization into the constructor, so that it is easier to see